### PR TITLE
type.dd: minor touchup

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by WalterBright** [comment]
Something is terribly wrong with DAutoTest. Again, it fails somewhere, somehow on type.dd. But the voluminous log file has no error messages whatsoever referring to type.dd, let alone any line in it where it went wrong.

**Comment by MetaLang** [comment]
I think (???) this is the offender:

```
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on .generated/dlangspec.log.
posix.mak:550: recipe for target '/dev/shm/dtest/work/repo/dlang.org/web/dlangspec.pdf' failed
make: *** [/dev/shm/dtest/work/repo/dlang.org/web/dlangspec.pdf] Error 1
fatal: 'refs/ae-sys-d-cache/website-854fe828d188bf06f6d59414a7bfd08c024e96d7-0154801fd798f1f652fc3f8292385afa' - not a valid ref
```

But I have no clue why the ref is invalid. @CyberShadow

**Comment by wilzbach** [comment]
> But the voluminous log file has no error messages whatsoever referring to type.dd, let alone any line in it where it went wrong.

Just scroll to the end:

```

! Improper alphabetic constant.
<to be read again> 
                   \xFF 
l.4815 \end{tabularx}
                     
? 
! Emergency stop.
<to be read again> 
                   \xFF 
l.4815 \end{tabularx}
                     
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on .generated/dlangspec.log.
posix.mak:550: recipe for target '/dev/shm/dtest
```

File: spec/type.dd [hunk 1]
```diff
@@ -30,9 +38,9 @@ $(H2 $(LEGACY_LNAME2 Basic Data Types, basic-data-types, Basic Data Types))
     $(TROW $(D cfloat), $(D float.nan+float.nan*1.0i), a complex number of two float values)
     $(TROW $(D cdouble), $(D double.nan+double.nan*1.0i), complex double)
     $(TROW $(D creal), $(D real.nan+real.nan*1.0i), complex real)
-    $(TROW $(D char), $(D 0xFF), unsigned 8 bit (UTF-8 code unit))
-    $(TROW $(D wchar), $(D 0xFFFF), unsigned 16 bit (UTF-16 code unit))
-    $(TROW $(D dchar), $(D 0x0000FFFF), unsigned 32 bit (UTF-32 code unit))
+    $(TROW $(D char), $(D '\xFF'), unsigned 8 bit (UTF-8 code unit))
+    $(TROW $(D wchar), $(D '\uFFFF'), unsigned 16 bit (UTF-16 code unit))
+    $(TROW $(D dchar), $(D '\U0000FFFF'), unsigned 32 bit (UTF-32 code unit))
```
Comment (wilzbach) [inline spec/type.dd]: LaTeX offers `\textbackslash`, but AFAICT it hasn't been used before, so you would need to do sth. like:

spec/latex.ddoc
```
BACKSLASH=\textbackslash
```

spec/dlang.org.ddoc and ebook.ddoc:
```
BACKSLASH=\
```

(I don't like the Ddoc setup either)

Comment (WalterBright) [reply spec/type.dd]: Curiously, `\x...` is used in lex.dd

**Comment by CyberShadow** [comment]
> But I have no clue why the ref is invalid. @CyberShadow

That's an unrelated message coming from git, ignore it.

**Comment by WalterBright** [comment]
@CyberShadow Can you please help with this? I have no idea how to get it to pass DAutoTest, the log files leave me clueless.

**Comment by wilzbach** [comment]
> The spec claims that code that compiles does not compile. This is the kind of stuff that would be easily detected automatically.

Yeah - one just has to annotate the entire spec: https://github.com/dlang/dlang.org/pull/2120

**Comment by CyberShadow** [comment]
Thanks all!

**Pull request merged**